flow/project.py

```diff
@@ -441,6 +441,206 @@ def __call__(self, job=None):
             return self._cmd.format(job=job)
 
 
+class FlowGroup(object):
+    """A FlowGroup represents a subset of a workflow for any given job.
+
+    Any :class:`FlowGroup` is associated with a group of
+    :class:`FlowOperation`s. The command (cmd) is by default a modified
+    :func:`run` command, but can be overwritten with any function that takes a
+    job and operations argument.
+
+    :param name:
+        The name of the group to be used when calling from the commandline.
+    :type name:
+        :class:`str`
+    :param operations:
+        The list of operation names associated with the group.
+    :type operations:
+        :class:`list` of :class:`FlowOperation`
+    :param cmd:
+        The command to execute group; should be a function of job, and
+        operations.
+    :type cmd:
+        str or callable
+    :param directives:
+        A dictionary of additional parameters that provide instructions on how
+        to execute this operation, e.g., specifically required resources.
+    :type directives:
+        :class:`dict`
+    :param options:
+        A string of options to append to the output of the object's call method.
+        This lets options like --num_passes to be given to a group.
+    :type options:
+        :class:`str`
+    """
+
+    def __init__(self, name, operations=None, cmd=None, directives=None,
+                 options=None):
+        self.name = name
+        if options is None:
+            self.options = ""
+        else:
+            self.options = options
+        if operations is None:
+            self.operations = []
+        else:
+            self.operations = operations
+        if cmd is None:
+            def cmd(operations, job=None):
+                if job is None:
+                    return "run -o {}".format(' '.join(operations))
+                return "run -j {} -o {}".format(job, ' '.join(operations))
+            self._cmd = cmd
+        else:
+            self._cmd = cmd
+        self.directives = None
+
+    def __call__(self, job=None):
+        # Get string form of command
+        if callable(self._cmd):
+            return self._cmd(self.operations, job) + ' ' + self.options
+        else:
+            return self._cmd.format(self.operations, job) + ' ' + self.options
+
+    def add_operation(self, operation):
+        self.operations.append(operation)
+
+
+class JobGroup(FlowGroup):
```

Comment:
I'm a bit concerned about the code duplication here. In what way does a "group-operation" differ from a standard operation that we need to introduce this class?